### PR TITLE
MGMT-3911: Fixing 'belongs-to-majority-group' validation test.

### PR DESCRIPTION
**Pull request by ybettan**
Signed-off-by: Yoni Bettan <ybettan@redhat.com>

### CODE REVIEW

File: subsystem/metrics_test.go [hunk 1]
```diff
@@ -588,6 +594,34 @@ var _ = Describe("Metrics tests", func() {
...
+
+			// check generated metrics
+			metricCounter := getValidationMetricCounter(clusterID, models.HostValidationIDBelongsToMajorityGroup, hostValidationChangedMetric)
+			// this specific case can create a short timeframe in which another host is failing on that validation and will
+			// be later fixed by the next refresh status cycle because generating a full mesh connectivity isn't an atomic
+			// action, therefore, in this test we will check that at least the expected failing host is failing but not fail
+			// the test if other hosts fails as well.
+			Expect(metricCounter >= 1).To(BeTrue())
+			metricsDeregisterCluster(ctx, clusterID)
+			assertValidationMetricCounter(clusterID, models.HostValidationIDBelongsToMajorityGroup, hostValidationFailedMetric, 1)
```
Comment (ori-amizur) [inline subsystem/metrics_test.go:622]: There is a possibility that h2 will still be in failed state.  Is it OK for this assertion?

File: subsystem/metrics_test.go [hunk 2]
```diff
@@ -588,6 +594,34 @@ var _ = Describe("Metrics tests", func() {
...
+		})
```
Comment (ori-amizur) [inline subsystem/metrics_test.go:623]: There is a possibility that h2 will still be in failed state. Is it OK for this assertion?

Comment (ybettan) [reply subsystem/metrics_test.go:625]: Fixed.

**Pull request merged**